### ISSUE TITLE
Lilypond point and click

### ISSUE THREAD

**Issue by AljoschaMeyer** [izuzak/atom-pdf-view#141]
Hey there,

would it be possible to add support for point and click for [lilypond](http://lilypond.org/)? Lilypond is to traditional music notation software what latex is to WYSIWYG editors. It has a [point and click feature](http://lilypond.org/doc/v2.18/Documentation/usage/point-and-click) which lets you click on the generated PDF and takes you to the correct line in the source file.

This is basically the same thing as the SyncTeX support which pdf-view already offers. To quote from the [point and click documentation](http://lilypond.org/doc/v2.18/Documentation/usage/configuring-the-system-for-point-and-click#enabling-point-and-click):

> When this functionality is active, LilyPond adds hyperlinks to the PDF file. These hyperlinks are sent to a ‘URI helper’ or a web-browser, which opens a text-editor with the cursor in the right place.
>
> To make this chain work, you should configure your PDF viewer to follow hyperlinks using the ‘lilypond-invoke-editor’ script supplied with LilyPond.

I currently see two options: Either add support directly to pdf-view (just as with SyncTeX), or add some general API to pdf-view which lets other packages subscribe for link-activations.

Best regards,
Aljoscha

**Comment by izuzak** [comment]
Hey @AljoschaMeyer -- thanks for opening an issue! That sounds like it would be a neat and useful feature, and it seems like it should be possible to do. That said, I probably won't have time to look into that myself in the near future, so if someone else wants to try and add support for that -- I'd be happy to review the pull request.

**Comment by AljoschaMeyer** [comment]
I don't have the time to implement generalized uri click support either, but I could simply add some lilypond specific code analogous to synctex. Would you accept such a pull request?

**Comment by izuzak** [comment]
> I don't have the time to implement generalized uri click support either, but I could simply add some lilypond specific code analogous to synctex. Would you accept such a pull request?

Yeah, I believe I would if it doesn't cause problems for any of the existing stuff.

**Comment by AljoschaMeyer** [comment]
Ok, I took a look at the synctex implementation and there's a slight problem. I assumed that synctex was based on links, but it is actually a program which takes raw coordinates as input. This makes the implementation for lilypond support quite different from the synctex support.

**Comment by aiomaster** [comment]
@AljoschaMeyer Did you work at this feature somewhere? I would also be interested in this feature cause it would make atom more usefull for lilypond editing.
Or do you have an idea where to start implementing it?

**Comment by AljoschaMeyer** [comment]
I spent some time trying to write a pdf-viewer plugin for atom with support for annotations (e.g. links). But working with pdf.js was pretty horrible, since it predates javescript modules and e.g. uses a custom, make-file based module system. I did not manage to successfully display pdfs in atom with annotation support. At some point I [embedded the default pdfjs viewer in an iframe](https://github.com/AljoschaMeyer/atom-pdfjs). That gives you support for links (and e.g. the ability to select text). But integrating that with atom (keyboard shortcuts via the atom options, theming etc.) might be more effort than using the pdf-js api for building a suitable viewer plugin. Still, for quick-and-dirty point-and-click support, this might be the way to go.
